Microsoft.Toolkit.Graph.Providers/Graph.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Graph.Providers
+{
+    /// <summary>
+    /// Creates a new provider instance for the provided configuration and sets the GlobalProvider.
+    /// </summary>
+    public static class Graph
+    {
+        /// <summary>
+        /// Gets the Graph Config property value.
+        /// </summary>
+        /// <param name="target">
+        /// The target object to retrieve the property value from.
+        /// </param>
+        /// <returns>
+        /// The value of the property on the target.
+        /// </returns>
+        public static IGraphConfig GetConfig(DependencyObject target)
+        {
+            return (IGraphConfig)target.GetValue(ConfigProperty);
+        }
+
+        /// <summary>
+        /// Sets the GraphConfig property value.
+        /// </summary>
+        /// <param name="target">
+        /// The target object to set the value on.
+        /// </param>
+        /// <param name="value">
+        /// The value to apply to the target property.
+        /// </param>
+        public static void SetConfig(DependencyObject target, IGraphConfig value)
+        {
+            target.SetValue(ConfigProperty, value);
+        }
+
+        /// <summary>
+        /// Identifies the Config dependency property.
+        /// </summary>
+        /// <returns>
+        /// The identifier for the Config dependency property.
+        /// </returns>
+        public static readonly DependencyProperty ConfigProperty =
+            DependencyProperty.RegisterAttached("Config", typeof(IGraphConfig), typeof(Page), new PropertyMetadata(null, OnConfigChanged));
```

Comment:
```suggestion
        public static IGraphConfig GetConfig(ResourceDictionary target)
        {
            return (IGraphConfig)target.GetValue(ConfigProperty);
        }

        /// <summary>
        /// Sets the GraphConfig property value.
        /// </summary>
        /// <param name="target">
        /// The target object to set the value on.
        /// </param>
        /// <param name="value">
        /// The value to apply to the target property.
        /// </param>
        public static void SetConfig(ResourceDictionary target, IGraphConfig value)
        {
            target.SetValue(ConfigProperty, value);
        }

        /// <summary>
        /// Identifies the Config dependency property.
        /// </summary>
        /// <returns>
        /// The identifier for the Config dependency property.
        /// </returns>
        public static readonly DependencyProperty ConfigProperty =
            DependencyProperty.RegisterAttached("Config", typeof(IGraphConfig), typeof(Graph), new PropertyMetadata(null, OnConfigChanged));
```

They `typeof` at the bottom is the type of the owner of the Attached Property which is still our `Graph` component.

But for the Get/Set we can force the attached property to only be attachable to that specific type (ResourceDictionary).  Then we should be able to have the same initialization pattern by having folks put it in App.xaml:

```xml
    <Application.Resources>
        <ResourceDictionary>
            <providers:Graph.Config>
                <providers:MockConfig TestProp="Hello" TestProp2="World" />
            </providers:Graph.Config>
        </ResourceDictionary>
    </Application.Resources>
```

The MockConfig object here was properly initialized on startup with the values of the properties we'd need. 🙂

Reply:
This is great! I love having the declaration in the App.xaml. It just feels right lol